ocaml/xapi/xapi_host.ml

```diff
@@ -1418,19 +1413,18 @@ let reset_server_certificate ~__context ~host =
   in
   let dns_names = Networking_info.dns_names () in
   let ips = [ip] in
-  Gencertlib.Selfcert.host ~name ~dns_names ~ips xapi_ssl_pem ;
-  (* Reset stunnel to try to restablish TLS connections *)
-  Xapi_mgmt_iface.reconfigure_stunnel ~__context ;
-  (* Delete records of the server certificate in this host *)
-  let expr =
-    Db_filter_types.(Eq (Field "host", Literal (Ref.string_of self)))
-  in
-  Db.Certificate.get_refs_where ~__context ~expr
-  |> List.iter (fun self -> Db.Certificate.destroy ~__context ~self)
-
-let emergency_reset_server_certificate ~__context =
-  let host = Helpers.get_localhost ~__context in
-  reset_server_certificate ~__context ~host
+  Gencertlib.Selfcert.host ~name ~dns_names ~ips xapi_ssl_pem
+
+let reset_server_certificate ~__context ~host =
+  let dbg = Context.string_of_task __context in
+  let write_cert_fs () = _new_host_cert ~dbg in
+  replace_host_certificate ~__context ~host write_cert_fs
+
+let emergency_reset_server_certificate ~(__context : 'a) =
```

Comment:
Does this emergency call still work on a pool member if the master connection is broken due to problems with the certificate?

Reply:
I believe so, I just tried it out (Host.reset_server_certificate is of course expected to fail, but the emergency call still works)

Reply:
It does not modify the database, this means that when using the call the filesystem and the database will be out of sync. There is work incoming to synchronize them on xapi startup.

The call affect the certificate offered to external API clients, and therefore it cannot break pool operations.